servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -1801,6 +1802,33 @@ public final Publisher<T> whenCancel(Runnable onCancel) {
         return beforeCancel(onCancel);
     }
 
+    /**
+     * Validate the outstanding demand ({@link Subscription#request(long)} minus
+     * {@link Subscriber#onNext(Object)}) does not go negative.
+     * @return The new {@link Publisher}.
+     * @see #validateDemand(ObjLongConsumer, LongBinaryConsumer)
+     */
+    public final Publisher<T> validateDemand() {
+        return new ValidateDemandPublisher<>(this, (demand, next) -> { }, (n, demand) -> { });
+    }
+
+    /**
+     * Validate the outstanding demand ({@link Subscription#request(long)} minus
+     * {@link Subscriber#onNext(Object)}) does not go negative.
+     * <p>
+     * ReactiveStreams specification allows {@link Subscriber} and {@link Subscription} to be concurrent, therefore the
+     * arguments of this function maybe invoked concurrently.
+     * @param onNextConsumer Called when {@link Subscriber#onNext(Object)} is invoked with first argument is the signal
+     * and second argument is demand (after decrementing).
+     * @param requestConsumer Called when {@link Subscription#request(long)} is invoked with first argument of {@code n}
+     * and second argument of current demand (after incrementing). This method may only be called if {@code n} is valid.
```

Comment:
Consider clarifying how these functions can be used. Are those only to observe or can they also throw an exception?